src/composables/useGetThreadId.ts

```diff
@@ -4,24 +4,11 @@
  */
 
 import { createSharedComposable } from '@vueuse/core'
-import { computed, ref } from 'vue'
-import { useRoute, useRouter } from 'vue-router'
+import { ref } from 'vue'
 
 /**
  * Shared composable to get threadId of current thread in conversation
  */
 export const useGetThreadId = createSharedComposable(function() {
-	const router = useRouter()
-	const route = useRoute()
-
-	if (router) {
-		return computed<number>({
-			get: () => route.query.threadId ? Number(route.query.threadId) : 0,
-			set: (value: number) => {
-				router.push({ query: { ...route.query, threadId: value !== 0 ? value : undefined } })
-			},
-		})
-	} else {
-		return ref(0)
-	}
```

Comment:
Instead of duplicating router synchronisation in components, we can define a ref here, and when sync it with router.

Then, on the one hand, it will be just a ref, not vue-router state. And on the other hand - changing it would change the route.

Reply:
Seems to be working. Migrated useGetToken to useRouteParams as well, so both updates are now in sync